src/vault/client.clj

```diff
@@ -241,7 +302,30 @@
                       :as :json})]
       (log/debug "Deleted secret" path)
       (cache/invalidate! cache path)
-      (= (:status response) 204))))
+      (= (:status response) 204)))
+
+  (create-token! [this] (create-token! this nil))
+
+  (create-token!
+    [this wrap-ttl]
+    (check-auth! token)
+    (let [response (api-request :post (str api-url "/v1/auth/token/create")
+                                {:headers (build-headers @token {:wrap-ttl wrap-ttl})
+                                 :accept :json
+                                 :as :json})]
+      (log/debug "Created token" (when wrap-ttl "with X-Vault-Wrap-TTL" wrap-ttl))
+      (when (= (:status response) 200)
+        (:body response))))
```

Comment:
In the long run it seems more useful to return some custom type for wrapped values. 🤔